### ISSUE TITLE
Allow Customizing Client and Server Config Separately in GUI

### ISSUE THREAD

**Issue by hammy275** [hammy3502/immersive-mc#444]
LAN hosting and world sharing mods force the config of the host onto everyone, when, in a lot of cases, the host may want others to be able to use a feature, but not want to use it themselves. 

By default, the configuration UI would look the same, but modify the client config. A button would allow you to edit the server config instead, which also would remove all the client-only options from the screen while active.

Basically depends on #307, since we'd be storing two config files now.

**Comment by hammy275** [comment]
Work for this is being done in the `config-refactor` branch. At the moment, the biggest aspect of work to be done is migrating things such that `ActiveConfig` and `ClientActiveConfig` are properly split, instead of everything using `ClientActiveConfig`.

**Comment by hammy275** [comment]
On this front, `ActiveConfig` and `ClientActiveConfig` are properly split now, and the config GUI allows customizing between the two. Things left:

- [x] Clean up config syncing while in-game such that the server config is only propagated if the server one is changed.
- [x] Massive amounts of testing

**Comment by hammy275** [comment]
Ready for merge alongside #307. Further tracking will be done there in terms of Minecraft versions it's propagated to.

**Comment by hammy275** [comment]
Downported to all the other versions, this is finally squashed!
